docs/blog/2019-07-23-google-sheets-gatsby-acroyoga-video-explorer/index.md

```diff
@@ -0,0 +1,428 @@
+
+```javascript
+module.exports = {
+  plugins: [
+    /* highlight-start */
+    {
+      resolve: `gatsby-plugin-material-ui`,
+      options: {
+        theme: {
+          // material-ui theme
+          palette: {
+            primary: {
+              main: "#BA3D3B", // or whatever colors you need
+            },
+          },
+        },
+      },
+    },
+    /* highlight-end */
+  ],
+}
+```
+
+## My Google Sheet Dataset and How to Access it
+
+Google sheets are pretty amazing if you think about it. While they might never be able to handle the millions of rows that Excel can, you can share and collaborate on them. There's tons of add-ons, macros, and scripts you can use to extend the default functionality. Finally, you can consume the data from a Google Sheet using [a fairly well-documented API](https://developers.google.com/sheets/api/). My Acroyoga videos data can be publicly accessed [here](https://docs.google.com/spreadsheets/d/1Qyn6530gveP7wnLHswBH0a_6ndTbWE65hvMWY17313Y/edit?usp=sharing). Here's what my row headers look like and several of my rows:
+
+![Google Sheets UI](./images/word-image.png)
+
+This is a list of Acroyoga videos from Youtube. I tagged these videos in column I (e.g. beginner, flow). The 'thumbnail' column is the Youtube video thumbnail while instructor_image are some dummy images I saved in a Google cloud storage bucket. The instructor_image column is not [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) here and I could find a way to store this separately in another sheet. This occasional lack of 'DRYness' is one potential downside of using a spreadsheet as a web application database.
+
+To access this data, I first must 'publish to the web' (under the file menu). This makes it available for consuming via API. Next I needed to register for an API key with Google. Finally, I can GET this data using a Google Sheets API method called [batchGet](https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets.values/batchGet). My API call looks like this: [`https://sheets.googleapis.com/v4/spreadsheets/1Qyn6530gveP7wnLHswBH0a_6ndTbWE65hvMWY17313Y/values:batchGet?ranges=export&majorDimension=ROWS&key=AIzaSzC1XWLfbg_9cbbq6dv-eFROFVDpfp2XhxE`](https://sheets.googleapis.com/v4/spreadsheets/1Qyn6530gveP7wnLHswBH0a_6ndTbWE65hvMWY17313Y/values:batchGet?ranges=export&majorDimension=ROWS&key=AIzaSzC1XWLfbg_9cbbq6dv-eFROFVDpfp2XhxE)
```

Comment:
https://gist.github.com/kpennell/82484a220c0e56f40f98c4bcaec87d9c I put the gist here.

Reply:
@gillkyle Sorry, confusing. The API key is wrong on purpose. Does the Gist work? I'm new to co-editing posts on github. What's the easiest way for me to edit the post without causing a bunch more git hassles?

Reply:
The gist is great! Thanks!

And no worries at all, you can click on the 3 dot menu next to the file name which takes you to this link: 
https://github.com/kpennell/gatsby/edit/master/docs/blog/2019-07-23-google-sheets-gatsby-acroyoga-video-explorer/index.md?pr=%2Fgatsbyjs%2Fgatsby%2Fpull%2F15973

Then you can edit it and make a commit right there 🙂 

Since you made all those other edits I'll just make this change real fast myself so you don't have to worry about it.